doc/source/building/macosx.rst

```diff
@@ -49,6 +37,13 @@ See `this site <http://r.research.att.com/tools/>`__ for the most recent links.
 Unless you are building from released source packages, the `Cython
 <http://cython.org/>`__ compiler is also needed.
 
+BLAS/LAPACK Installation
+------------------------
+
+You will also need to install a library providing the BLAS and LAPACK
+interfaces. ATLAS, OpenBLAS, and MKL all work. OpenBLAS can be installed
+via `Homebrew <https://brew.sh/>`.
```

Comment:
Package manager comment as before. Can we give a link to OpenBlas, and instructions on how to build it? Or if we don't give detailed build instructions for OpenBlas, then any specific flags required to get it to work with scipy.

Reply:
I don't think that's a good idea. Homebrew for most users *just works*, and building a BLAS/LAPACK from source is very much nontrivial. Users who dislike a package manager will know enough to search for themselves. The default here should be the easiest option.

Reply:
I've just tried making OpenBlas. There was one minor hurdle (MACOSX_DEPLOYMENT_TARGET needs updating in `Makefile.system`), but apart from that it just builds with a `make`. For those of us who are interested - are there any specific things that need to be configured to use this version of OpenBlas to build scipy?

Reply:
IIRC you'lll want to build with `OPENBLAS_NUM_THREADS=1` (see https://github.com/xianyi/OpenBLAS/wiki/faq#multi-threaded). That's all I think

Reply:
If that's the case then the instructions for an OpenBlas would be:

1) Have a gfortran (w link)
2) Download OpenBlas
3) export OPENBLAS_NUM_THREADS=1
4) make
5) sudo make install

I may be biased, but but that doesn't seem overly difficult compared to having to install homebrew, and fiddling around with that. Does the same BLAS have to be used for numpy and scipy?

Reply:
> Does the same BLAS have to be used for numpy and scipy?

> I may be biased, but but that doesn't seem overly difficult compared to having to install homebrew, and fiddling around with that

I think it is more complicated for the beginning user - there's a lot of things that can go wrong there. Also, I'd prefer to minimize the variation in install configs (gcc and openblas versions, local build configs due to different compilers on different macOS versions, etc.) - issues are easier to debug if the devs and users have the exact same binaries on their machines.

Reply:
Speaking as someone who used to put some energy into avoiding package managers, I agree with Ralf - I think Homebrew is the most straightforward choice for beginning users.    I've been using Homebrew on a couple of laptops and a few desktops for about 5 years now, with few problems.

It would be good to link to some pages for Compiling OpenBLAS for Numpy / Scipy though.

Reply:
I agree with what's already been said here. A lot can go wrong when beginning users try to compile BLAS/LAPACK from source. If they can't use brew, maybe they can use conda (or maybe conan?). If we want to include docs for building dependencies from source at all, we should make sure beginners understand that that's a fallback for them.

I'm not sure of the extent to which we need to document install instructions for third party dependencies though. OpenBLAS, for example, already maintains up-to-date instructions that they maintain for how to build and install from source. Beginners should use a package manager. Anyone else will get better information from the website for the package they are interested in.

Reply:
`OPENBLAS_NUM_THREADS` needs to be set at runtime, not a build time. It'd also probably be preferable for users running serial python code to leave it alone and let OpenBLAS use as many threads as it wants. This seems like the kind of thing we'd really want to document somewhere, but I have no idea where. It's not a build-time thing per se, unless you want to compile a BLAS that doesn't use parallelism at all.